### PR TITLE
Add support for inserting WaterML 2.0 ObservationProcess as procedure description 

### PR DESCRIPTION
**Pull request by CarstenHollmann**

### CODE REVIEW

File: coding/sos-v20/src/main/java/org/n52/sos/decode/AbstractGmlDecoderv321.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+		} else if (agmlt.isSetDescriptionReference()) {
+			// TODO
+		}
+		return abstractGML;
+
+	}
+
+	protected AbstractGML parseMetaDataProperty(AbstractGMLType agmlt, AbstractGML abstractGML) {
+		if (CollectionHelper.isNotNullOrEmpty(agmlt.getMetaDataPropertyArray())) {
+			// TODO
```
Comment (autermann) [inline coding/sos-v20/src/main/java/org/n52/sos/decode/AbstractGmlDecoderv321.java]: so these are not part of the pull request? :)

File: coding/sos-v20/src/main/java/org/n52/sos/decode/AbstractOmDecoderv20.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+	protected NamedValue<?> parseNamedValueType(NamedValuePropertyType namedValueProperty) throws OwsExceptionReport {
+        if (namedValueProperty.isSetNamedValue()) {
+            NamedValueType namedValue = namedValueProperty.getNamedValue();
+            NamedValue<?> sosNamedValue = parseNamedValueValue(namedValue.getValue());
+            ReferenceType referenceType = (ReferenceType) CodingHelper.decodeXmlObject(namedValue.getName());
+            sosNamedValue.setName(referenceType);
+            return sosNamedValue;
+        } else if (namedValueProperty.isSetHref()) {
+            NamedValue<?> sosNamedValue = new NamedValue<ReferenceType>();
```
Comment (autermann) [inline coding/sos-v20/src/main/java/org/n52/sos/decode/AbstractOmDecoderv20.java]: Why `<?>`? The actual type is right there...

File: coding/sos-v20/src/main/java/org/n52/sos/decode/AbstractOmDecoderv20.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        } else if (value instanceof ReferenceType) {
+            NamedValue<ReferenceType> namedValue = new NamedValue<ReferenceType>();
+            namedValue.setValue(new ReferenceValue((ReferenceType)value));
+            return namedValue;
+        } else if (value instanceof W3CHrefAttribute) {
+            NamedValue<W3CHrefAttribute> namedValue = new NamedValue<W3CHrefAttribute>();
+            namedValue.setValue(new HrefAttributeValue((W3CHrefAttribute)value));
+            return namedValue;
+        } else {
+            throw new UnsupportedDecoderInputException(this, xmlObject);
```
Comment (autermann) [inline coding/sos-v20/src/main/java/org/n52/sos/decode/AbstractOmDecoderv20.java]: We really need a `ValueVisitor`...

File: coding/sos-v20/src/main/java/org/n52/sos/decode/OmDecoderv20.java [hunk 4]
```diff
@@ -145,21 +136,19 @@ public Object decode(Object object) throws OwsExceptionReport {
...
         if (object instanceof OMObservationType) {
             return parseOmObservation((OMObservationType) object);
-        } else if (object instanceof NamedValuePropertyType) {
-            return parseNamedValueType((NamedValuePropertyType) object);
-        } else if (object instanceof NamedValuePropertyType[]) {
-            return parseNamedValueTypeArray((NamedValuePropertyType[]) object);
+//        } else if (object instanceof NamedValuePropertyType) {
+//            return parseNamedValueType((NamedValuePropertyType) object);
+//        } else if (object instanceof NamedValuePropertyType[]) {
+//            return parseNamedValueTypeArray((NamedValuePropertyType[]) object);
```
Comment (autermann) [inline coding/sos-v20/src/main/java/org/n52/sos/decode/OmDecoderv20.java]: Why keep this as a comment?

File: coding/wml-v20/src/main/java/org/n52/sos/decode/AbstractWmlDecoderv20.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ */
+package org.n52.sos.decode;
+
+public abstract class AbstractWmlDecoderv20 extends AbstractOmDecoderv20 {
+
+
+}
```
Comment (autermann) [inline coding/wml-v20/src/main/java/org/n52/sos/decode/AbstractWmlDecoderv20.java]: Looks kinda dispensable...

File: coding/wml-v20/src/main/java/org/n52/sos/decode/WmlObservationProcessDecoderv20.java [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+	public Set<String> getSupportedProcedureDescriptionFormats(String service, String version) {
+		if (SUPPORTED_TRANSACTIONAL_PROCEDURE_DESCRIPTION_FORMATS.containsKey(service)
+				&& SUPPORTED_TRANSACTIONAL_PROCEDURE_DESCRIPTION_FORMATS.get(service).containsKey(version)) {
+			return SUPPORTED_TRANSACTIONAL_PROCEDURE_DESCRIPTION_FORMATS.get(service).get(version);
+		}
+		return Collections.emptySet();
+	}
+
+	@Override
+	public Object decode(Object object) throws OwsExceptionReport, UnsupportedDecoderInputException {
```
Comment (autermann) [inline coding/wml-v20/src/main/java/org/n52/sos/decode/WmlObservationProcessDecoderv20.java]: Why not splitting this class up into 3 classes and get rid of this `Object` types?

**Comment by autermann** [comment]
Waiting for 4 spaces :)

File: coding/wml-v20/src/main/java/org/n52/sos/decode/AbstractWmlDecoderv20.java (hunk 5 above)
Comment (CarstenHollmann) [inline coding/wml-v20/src/main/java/org/n52/sos/decode/AbstractWmlDecoderv20.java]: @bpross-52n starts implementing WaterML 2.0 observation decoder. Maybe he will use this class.

File: coding/sos-v20/src/main/java/org/n52/sos/decode/AbstractOmDecoderv20.java (hunk 3 above)
Comment (CarstenHollmann) [inline coding/sos-v20/src/main/java/org/n52/sos/decode/AbstractOmDecoderv20.java]: Created a new issue #298
